src/Disciplina/Crypto/ByteArray.hs

```diff
@@ -0,0 +1,76 @@
+
+-- | Utilities for manipulating byte arrays.
+
+module Disciplina.Crypto.ByteArray
+       ( FromByteArray (..)
+       , hashBytesWithSalt
+       , hashBytes
+       ) where
+
+import Universum
+
+import Crypto.Error (maybeCryptoError)
+import qualified Crypto.Hash as Crypto
+import qualified Crypto.PubKey.Ed25519 as Ed25519
+import Data.ByteArray (ByteArray, ByteArrayAccess)
+import qualified Data.ByteArray as BA
+import qualified Data.Hashable as Hash
+import System.IO.Unsafe (unsafeDupablePerformIO)
+
+-- | Class of types, which can be reconstructed from a 'ByteArray',
+-- but only from a particular one.
+class ByteArrayAccess ba => FromByteArray ba where
+    fromByteArray :: ByteArrayAccess ba' => ba' -> Maybe ba
```

Comment:
So why have you ignored my two comments on `Maybe` vs `Either`?

Reply:
1. Less boilerplate is needed with `Maybe`
2. It still doesn't seem very good to have `String` as error, and I don't know which ADT to imagine there (see [exceptions.md](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md))
3. I realized that we'll want custom `Serialise` instances for hashes and signatures in the end, because we know their _sizes_, and casting them to bytestring before serialization is suboptimal (bytestrings are prepended with their size), so in the end (de)serialisation errors for them would be provided directly by their instances. 

Though last point is actually an argument in favor of removing `FromByteArray` class altogether :thinking: